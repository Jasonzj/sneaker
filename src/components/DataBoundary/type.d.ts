import { ReactElement } from 'react'
export type Props = {
  error: unknown
  info?: string | ReactElement
  loading?: {
    spinning: boolean
    isShowText?: boolean
    backGroundColor?: string
  }
}
