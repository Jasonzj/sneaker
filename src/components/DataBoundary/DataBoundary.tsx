import React from 'react'
import Loading from '../Loading'
import { Props } from './type'

const DataBoundary: React.FC<Props> = ({ error, info, loading = null, children }) => (
  <>
    {loading && <Loading {...loading} />}
    {error ? info : children}
  </>
)

export default DataBoundary
