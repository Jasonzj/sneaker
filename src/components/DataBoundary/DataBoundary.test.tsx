import { render, screen } from '@testing-library/react'
import DataBoundary from './DataBoundary'

const props = {
  error: false,
  info: <div>Search Fail!</div>,
  loading: {
    spinning: true,
  },
}

test('should take a snapshot', () => {
  const { asFragment } = render(
    <DataBoundary {...props}>
      <div>test</div>
    </DataBoundary>,
  )

  expect(asFragment()).toMatchSnapshot()
})

test('should correct display loading status', () => {
  const { rerender } = render(<DataBoundary {...props} />)

  expect(screen.getByTitle('loader')).toBeInTheDocument()

  rerender(<DataBoundary {...props} loading={undefined} />)

  expect(screen.queryByTitle('loader')).not.toBeInTheDocument()
})

test('should correct handle error', () => {
  const { rerender } = render(
    <DataBoundary {...props}>
      <div>test</div>
    </DataBoundary>,
  )

  expect(screen.getByText('test')).toBeInTheDocument()
  expect(screen.queryByText('Search Fail!')).not.toBeInTheDocument()

  rerender(
    <DataBoundary {...props} error={true}>
      <div>test</div>
    </DataBoundary>,
  )

  expect(screen.getByText('Search Fail!')).toBeInTheDocument()
  expect(screen.queryByText('test')).not.toBeInTheDocument()
})
