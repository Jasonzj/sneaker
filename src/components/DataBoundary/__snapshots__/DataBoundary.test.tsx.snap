// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="loader"
    title="loader"
  >
    <div
      class="warpper"
    >
      <div
        class="inner"
      />
    </div>
  </div>
  <div>
    test
  </div>
</DocumentFragment>
`;
